CMakeLists.txt

```diff
@@ -279,34 +292,40 @@ set(ALLOCATOR "" CACHE STRING
   "specify memory allocator to use. currently tcmalloc, tcmalloc_minimal, \
 jemalloc, and libc is supported. if not specified, will try to find tcmalloc, \
 and then jemalloc. If neither of then is found. use the one in libc.")
-if(ALLOCATOR)
-  if(${ALLOCATOR} MATCHES "tcmalloc(_minimal)?")
-    find_package(gperftools REQUIRED)
-    set(HAVE_LIBTCMALLOC ON)
-  elseif(${ALLOCATOR} STREQUAL "jemalloc")
-    find_package(JeMalloc REQUIRED)
-    set(HAVE_JEMALLOC 1)
-  elseif(NOT ALLOCATOR STREQUAL "libc")
-    message(FATAL_ERROR "Unsupported allocator selected: ${ALLOCATOR}")
-  endif()
-else(ALLOCATOR)
-  find_package(gperftools)
-  set(HAVE_LIBTCMALLOC ${gperftools_FOUND})
-  if(NOT gperftools_FOUND)
-    find_package(JeMalloc)
-  endif()
-  if(gperftools_FOUND)
-    set(ALLOCATOR tcmalloc)
-  elseif(JeMalloc_FOUND)
-    set(ALLOCATOR jemalloc)
-  else()
-    if(NOT FREEBSD)
-      # FreeBSD already has jemalloc as its default allocator
-      message(WARNING "tcmalloc and jemalloc not found, falling back to libc")
+# For now, we'll enforce libc for Windows builds. That's mostly because
+# the others aren't detected properly.
+if(NOT WIN32)
+  if(ALLOCATOR)
```

Comment:
can we just pass `-DALLOCATOR=libc` when building for WIN32, instead encoding this knowledge in cmake?

Reply:
That would work as well.